test/james.data.constraints.tests.pas

```diff
@@ -109,11 +109,11 @@ procedure TDataStreamTest.AsString;
   Buf := TMemoryStream.Create;
   Ss := TStringList.Create;
   try
-    Buf.WriteBuffer(TXT[1], Length(TXT) * SizeOf(Char));
+    Buf.WriteBuffer(AnsiString(TXT)[1], Length(TXT));
```

Comment:
I think we can forget `WriteBuffer` if it is impossible using "simple strings".
If this change penalizes performance, we can create a WriteBuffer/ReadBuffer wrapper for each compiler.